File: src/1394-find-lucky-integer-in-an-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
function findLucky(arr) {
  const map = new Map();

  for (const n of arr) {
    map.set(n, (map.get(n) || 0) + 1);
  }

  let lucky = -1;

  for (const [n, occurs] of map) {
    if (n === occurs) {
      lucky = Math.max(n, lucky);
    }
  }

  return lucky;
}

const arr = [2, 2, 3, 4];

const output = findLucky(arr);

console.log(output); // eslint-disable-line no-console

File: src/1394-find-lucky-integer-in-an-array.py
def findLucky(arr):
    count_map = {}

    for n in arr:
        count_map[n] = count_map.get(n, 0) + 1

    lucky = -1

    for n, occurs in count_map.items():
        if n == occurs:
            lucky = max(n, lucky)
    return lucky

arr = [2, 2, 3, 4]
output = findLucky(arr)

print(output)
